stdlib/messaging/activemq-artemis/src/main/ballerina/src/artemis/message.bal

```diff
@@ -20,34 +20,44 @@ public type Message client object {
 
     private MessageType messageType = TEXT;
     private MessageConfiguration configuration;
+    private MessageContent | Error content;
+    private boolean created = false;
+    private Session session;
 
     public function __init(Session session, @untainted MessageContent data, MessageConfiguration? config = ()) {
+        self.session = session;
+        self.content = data;
         if (config is MessageConfiguration) {
             self.configuration = config;
         } else {
             self.configuration = {};
         }
+        if(!self.created) {
+            self.initCreation(data);
+        }
+    }
 
+    function initCreation(MessageContent data) {
         if (data is (string | json | xml | int | float | byte | boolean)) {
             self.messageType = TEXT;
         }
```

Comment:
Don't we need else part here?

Reply:
It sets the type to text if it is one of these and again these data types are checked to perform other functions.